### ISSUE TITLE
Don't break polymorphic type arguments in function which is a last breaking argument

### ISSUE THREAD

**Issue by xixixao** [prettier/prettier#2581]
from
```es6
const appIDs = createSelector(
  PubXURLParams.APP_IDS,
 (rawAppIDs): Array<AppID> => deserializeList(rawAppIDs),
);
```
to
```es6
const appIDs = createSelector(PubXURLParams.APP_IDS,
 (rawAppIDs): Array<
  AppID
> => deserializeList(rawAppIDs));
```
verified in playground.

**Comment by bakkot** [comment]
I get something [slightly different](https://prettier.io/playground/#%7B%22content%22%3A%22const%20appIDs%20%3D%20createSelector(%5Cn%20%20PubXURLParams.APP_IDS%2C%5Cn%20(rawAppIDs)%3A%20Array%3CAppID%3E%20%3D%3E%20deserializeList(rawAppIDs)%2C%5Cn)%3B%22%2C%22options%22%3A%7B%22printWidth%22%3A80%2C%22tabWidth%22%3A2%2C%22singleQuote%22%3Afalse%2C%22trailingComma%22%3A%22none%22%2C%22bracketSpacing%22%3Atrue%2C%22jsxBracketSameLine%22%3Afalse%2C%22parser%22%3A%22babylon%22%2C%22semi%22%3Atrue%2C%22useTabs%22%3Afalse%2C%22doc%22%3Afalse%2C%22ast%22%3Afalse%7D%7D) in the playground:

```js
const appIDs = createSelector(PubXURLParams.APP_IDS, (rawAppIDs): Array<
  AppID
> => deserializeList(rawAppIDs));
```

I think this is a duplicate of https://github.com/prettier/prettier/issues/2558.

**Comment by j-f1** [comment]
Will be fixed in #4228.
